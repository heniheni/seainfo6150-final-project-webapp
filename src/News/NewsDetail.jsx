import React from "react";
import styles from "./NewsDetail.module.css";

import HTMLText from "../Helper/HTMLText.jsx"


const NewsDetail = (props) => {
  let NewsDetailContent;

 
    NewsDetailContent = (
      <div>
         <article>
           <div>
             <h1 className={styles.stylingTitle}>{props.newsDetail.title}</h1>
             <div className={styles.stylingDate}>
                    <time dateTime={props.newsDetail.datetime} >{props.newsDetail.datePublished}</time>
                  </div>
           </div>
           <div className={styles.coverContainer}>
           <img className={styles.stylingNewsCover}
                  src={props.newsDetail.newsImageUrl}
                  alt="News Cover"
                />
           </div>
                
        
                  <div>
                    <h4 className={styles.shortTextStyling}>{props.newsDetail.shortText}</h4>
                  </div>
                  {/* <div className={styles.contentContainer}>
                    <p className={styles.contentStyling}>{props.newsDetail.detailedText}</p>
                  </div> */}

                  <div className={styles.contentContainer}>
                  <HTMLText text={props.newsDetail.detailedText} />
                </div>

                  
               
              </article>
      </div>
    );
  

  return <div>{NewsDetailContent}</div>;
};

export default NewsDetail;
