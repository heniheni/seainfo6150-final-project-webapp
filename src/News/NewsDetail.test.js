import React from "react";
import NewsDetail from "./NewsDetail.jsx";

describe("NewsDetail tests", () => {
    
    var newsDetail = '{ "newsId": "match-1-mi-rcb","title": "dummy title","datetime": "2021-04-09T15:30:00.000Z","datePublished": "April 9th 2021, 9:30 pm","newsImageUrl": "https://resources.platform.iplt20.com/photo-resources/2021/04/09/d4189aa6-da1a-41af-9132-53b23cdde606/Match01-_00532-1-.JPG?width=2000&height=1333","shortText": "Harshal Patel starred with the ball and scalped a five-wicket haul.","detailedText": "Test Text"}';


    var json = JSON.parse(newsDetail);
    it("renders correctly", () => {
      const { container } = render(<NewsDetail newsDetail={json} />);
      expect(container).toMatchSnapshot();
    });

});