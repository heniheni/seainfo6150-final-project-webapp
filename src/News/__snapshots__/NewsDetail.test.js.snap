// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsDetail tests renders correctly 1`] = `
<div>
  <div>
    <div>
      <article>
        <div>
          <h1
            class="stylingTitle"
          >
            dummy title
          </h1>
          <div
            class="stylingDate"
          >
            <time
              datetime="2021-04-09T15:30:00.000Z"
            >
              April 9th 2021, 9:30 pm
            </time>
          </div>
        </div>
        <div
          class="coverContainer"
        >
          <img
            alt="News Cover"
            class="stylingNewsCover"
            src="https://resources.platform.iplt20.com/photo-resources/2021/04/09/d4189aa6-da1a-41af-9132-53b23cdde606/Match01-_00532-1-.JPG?width=2000&height=1333"
          />
        </div>
        <div>
          <h4
            class="shortTextStyling"
          >
            Harshal Patel starred with the ball and scalped a five-wicket haul.
          </h4>
        </div>
        <div
          class="contentContainer"
        >
          <div
            class="detailedTextStyling"
          >
            Test Text
          </div>
        </div>
      </article>
    </div>
  </div>
</div>
`;
